lib/ansible/modules/network/netact/netact_cm_command.py

```diff
@@ -0,0 +1,364 @@
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+racclimx = '/opt/oss/bin/racclimx.sh'
+
+
+def run_module():
+    """
+    Main module where option are handled and command is executed
+    :return:
+    """
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        operation=dict(type='str', required=True,
+                       aliases=['op'],
+                       choices=['Upload', 'Provision', 'Import',
+                                'Export', 'Provision_Mass_Modification']),
+        opsname=dict(type='str', required=False, aliases=['opsName']),
+        dn=dict(type='str', required=False, aliases=['DN']),
+        ws=dict(type='str', required=False, aliases=['WS']),
+        mr=dict(type='str', required=False, aliases=['MR']),
+
+        planname=dict(type='str', required=False, aliases=['planName']),
+        typeoption=dict(type='str', required=False, aliases=['type'],
+                        choices=['plan', 'actual', 'reference', 'template', 'siteTemplate']),
+        fileformat=dict(type='str', required=False, choices=['CSV', 'RAML2', 'XLSX'],
+                        aliases=['fileFormat']),
+        filename=dict(type='str', required=False, aliases=['fileName']),
+
+        createBackupPlan=dict(type='str', required=False, choices=['true', 'false'],
+                              aliases=['createBackupPlan']),
+        backupPlanName=dict(type='str', required=False, aliases=['backupPlanName']),
+        inputFile=dict(type='str', required=False),
+
+        verbose=dict(type='str', required=False),
+        extra_opts=dict(type='str', required=False)
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        original_message='',
+        message=''
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
+        return result
+
+    # manipulate or modify the state as needed (this is going to be the
+    # part where your module will do what it needs to do)
+
+    operation = module.params.get('operation')
+    if not operation:
+        module.fail_json(msg='Operation not defined', **result)
+
+    opsname = module.params.get('opsname')
```

Comment:
A validation on values of each input parameter can be good addition here, if there it available.

Reply:
Checking of some input values can be quite complex task and I not what to repeat that work in module side. This parameter checking already implement on racclimx.sh

Reply:
Yes failure is sent back to module. 
Error is this kind
... "msg": "Invalid argument: -planNamedd", "rc": 255, ...